test/integration/targets/apt/tasks/repo.yml

```diff
@@ -40,6 +40,106 @@
         state: absent
         allow_unauthenticated: yes
 
+# https://github.com/ansible/ansible/issues/77969
+- block:
+  - name: Pin the preferred version of foo
+    copy:
+      content: |-
+              Package: foo
+              Pin: version 1.0.0
+              Pin-Priority: 1001
+      dest: /etc/apt/preferences.d/foo
+
+  - name: Install unspecified version
+    apt:
+      name: foo
+      state: latest
+    register: apt_result
+
+  - name: Check install with dpkg
+    shell: dpkg-query -l foo
+    register: dpkg_result
+
+  - name: Check if install was successful
+    assert:
+      that:
+        - "apt_result is success"
+        - "dpkg_result is success"
+        - "'1.0.0' in dpkg_result.stdout"
+
+  - name: Remove foo between tests
+    apt:
+      name: foo
+      state: absent
+
+  - name: Test preferring pinned version when possible with >=
+    apt:
+      name: foo>=1.0.0
+      state: latest
+    register: apt_result
+
+  - name: Check install with dpkg
+    shell: dpkg-query -l foo
+    register: dpkg_result
+
+  - name: Check if install was successful
+    assert:
+      that:
+        - "apt_result is success"
+        - "dpkg_result is success"
+        # NOTE: latest version is not installed
```

Comment:
Not sure if this is desired behavior. Maybe instead the documentation should indicate `>=`/`=` will ignore any apt preferences instead of trying to support both.

Reply:
Removed all of that, just fixing installing an unspecified version now.